Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp

```diff
@@ -123,8 +125,20 @@ int RGroupDecomposition::add(const ROMol &inmol) {
       // 2 RGroup attachments (see https://github.com/rdkit/rdkit/pull/4002)
 
       // match the reduced representation:
-      std::vector<MatchVectType> baseMatches =
-          SubstructMatch(mol, *core.second.matchingMol, sssparams);
+      std::vector<MatchVectType> baseMatches;
+      bool useNormalMatch = true;
+      if (params().allowTautomerCore) {
+        auto tautomerQuery = core.second.getMatchingTautomerQuery();
+        if (tautomerQuery != nullptr) {
+          // query atom indices from the tautomer query are the same as the
+          // template matching molecule
+          baseMatches = tautomerQuery->substructOf(mol, sssparams);
```

Comment:
if tautomerQuery is nullptr, no matches should be performed,  shouldn't we call back to the original query?

Reply:
it doesn't look to me like `useNormalMatch` is actually doing anything here.
The `else` on line 138 is only evaluated if `params().doTautomers` is false, but `useNormalMatch` only gets set to false if `params().doTautomers` is true.

Am I missing something obvious or do you need to get rid of the `else` on l138?

Reply:
Clearly I write very confusing code, but I do believe that this logic is necessary.

`useNormalMatch` gets set to false if `params().doTautomers` is true *and* `tautomerQuery != nullptr`.  `useNormalMatch` is needed for the cases where `params().doTautomers` is true *and* `tautomerQuery == nullptr`- which will happen if the core cannot be Kekulized.